### ISSUE TITLE
{false,null,true} can't be used as name prefixes

### ISSUE THREAD

**Issue by goosmurf** [open-policy-agent/opa#1582]
# Expected Behavior

These statements should define rules:

false_asdf = "hello"
null_asdf = "hello"
true_asdf = "hello"

# Actual Behavior

$ opa run
OPA 0.12.1 (commit 2799e148, built at 2019-07-06T15:26:32Z)

Run 'help' to see a list of commands.

```
> false_asdf = "hello"
|
1 error occurred: 2:1: rego_parse_error: no match found
        false_asdf = "hello"
```

**Comment by tsandall** [comment]
@goosmurf thanks for reporting this. The issue seems to be that [scalars are ordered before vars in the grammar](https://github.com/open-policy-agent/opa/blob/master/ast/rego.peg#L170) and so the parser successfully consumes true/false/null and then tries to continue parsing from the next character (e.g., `_` in your examples) and fails (because it's going to expect whitespace or an infix operator like =, ==, etc.) We should be able to fix this entirely inside of the grammar.

**Comment by tsandall** [comment]
Fixed by #1611
